### PR TITLE
Add recipe for libvulkan-loader package

### PR DESCRIPTION
**Pull request by traversaro**
This adds the recipe for the `vulkan-loader` package. 

Repo: https://github.com/KhronosGroup/Vulkan-Loader .
Repology page: https://repology.org/project/vulkan-loader/versions .

The package name starts with `lib` given the discussion in https://github.com/conda-forge/staged-recipes/pull/19764#pullrequestreview-1085836936 .

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge/help-c-cpp The PR is ready for review.

File: recipes/vulkan-loader/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# that are available for download at https://vulkan.lunarg.com/sdk/home) are the sdk-*** ones
+# See https://github.com/KhronosGroup/Vulkan-Headers#version-tagging-scheme
+source:
+  url: https://github.com/KhronosGroup/Vulkan-Loader/archive/refs/tags/sdk-{{ version }}.tar.gz
+  sha256: fa2078408793b2173f174173a8784de56b6bbfbcb5fb958a07e46ef126c7eada
+
+build:
+  number: 0
+  ignore_run_exports_from:
+    - libxcb
```
Comment (hmaarrfk) [inline recipes/vulkan-loader/meta.yaml:18]: why?

File: recipes/vulkan-loader/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - if not exist %LIBRARY_BIN%\\vulkan-1.dll exit 1  # [win]
+
+about:
+  home: https://github.com/KhronosGroup/Vulkan-Loader
+  license: Apache-2.0
+  license_file: LICENSE.txt
+  summary: Khronos official Vulkan ICD desktop loader for Windows, Linux, and MacOS.
+
+extra:
+  feedstock-name: vulkan-loader
```
Comment (hmaarrfk) [inline recipes/vulkan-loader/meta.yaml:47]: why this distinction?

**Comment by hmaarrfk** [comment]
So OSX seems to install a few things in `${PREFIX}/loader` is that expected?

```
-- Installing: $PREFIX/loader/vulkan.framework
-- Installing: $PREFIX/loader/vulkan.framework/Resources
-- Installing: $PREFIX/loader/vulkan.framework/Versions
-- Installing: $PREFIX/loader/vulkan.framework/Versions/A
-- Installing: $PREFIX/loader/vulkan.framework/Versions/A/Resources
-- Installing: $PREFIX/loader/vulkan.framework/Versions/A/Resources/Info.plist
-- Installing: $PREFIX/loader/vulkan.framework/Versions/A/vulkan
-- Installing: $PREFIX/loader/vulkan.framework/Versions/Current
-- Installing: $PREFIX/loader/vulkan.framework/vulkan
-- Installing: $PREFIX/lib/libvulkan.1.3.239.dylib
-- Installing: $PREFIX/lib/libvulkan.1.dylib
-- Installing: $PREFIX/lib/libvulkan.dylib
```

Otherwise, this looks great.

cc @rossant 

xref: https://github.com/conda-forge/cdt-builds/pull/56

**Comment by traversaro** [comment]
> So OSX seems to install a few things in `${PREFIX}/loader` is that expected?
> 
> ```
> -- Installing: $PREFIX/loader/vulkan.framework
> -- Installing: $PREFIX/loader/vulkan.framework/Resources
> -- Installing: $PREFIX/loader/vulkan.framework/Versions
> -- Installing: $PREFIX/loader/vulkan.framework/Versions/A
> -- Installing: $PREFIX/loader/vulkan.framework/Versions/A/Resources
> -- Installing: $PREFIX/loader/vulkan.framework/Versions/A/Resources/Info.plist
> -- Installing: $PREFIX/loader/vulkan.framework/Versions/A/vulkan
> -- Installing: $PREFIX/loader/vulkan.framework/Versions/Current
> -- Installing: $PREFIX/loader/vulkan.framework/vulkan
> -- Installing: $PREFIX/lib/libvulkan.1.3.239.dylib
> -- Installing: $PREFIX/lib/libvulkan.1.dylib
> -- Installing: $PREFIX/lib/libvulkan.dylib
> ```
> 
> Otherwise, this looks great.
> 
> cc @rossant 
> 
> xref: https://github.com/conda-forge/cdt-builds/pull/56

Thanks for noticing this, I guess there should be some option to disable the installation of the library as a macOS framework, I will check this.

**Comment by traversaro** [comment]
> $PREFIX/loader/vulkan.framework

I disable framework installation with a one-line patch.

File: recipes/vulkan-loader/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipes/vulkan-loader/meta.yaml:20]: At least the version that is being packaged here was just using the headers of xcb, without linking the library, see https://github.com/KhronosGroup/Vulkan-Loader/blob/sdk-1.3.239.0/CMakeLists.txt#L180 . So keeping the `run_exports` was creating warnings (or an error? I can't remember) during the build. I noticed that this just changed in the latest master in https://github.com/KhronosGroup/Vulkan-Loader/pull/1136, so probably we will need anyhow to get rid of this in the next release.

File: recipes/vulkan-loader/meta.yaml (hunk 2 above)
Comment (traversaro) [reply recipes/vulkan-loader/meta.yaml:50]: In the past I packaged several repositories called `<something>` that just contained a C/C++ library, so I initially created a feedstock called `lib<something>-feedstock`, and then more packages were added (such as `<something>-python`, etc etc), and it turned out to be confusing for contributors to understand why the feedstock was called `lib<something>-feedstock` while the original repo was called `<something>`, and the feedstock name could not be (easily) changed. So, I started to create all the feedstosk by using the original repo name, a bit like debian source packages are named. Anyhow, I do not feel strongly about this, so I am happy to just name the feedstock as well `libvulkan-loader`.

Comment (hmaarrfk) [reply recipes/vulkan-loader/meta.yaml:50]: thank you for explaining.

**Comment by hmaarrfk** [comment]
Looks good to me. Thanks!

**Review by hmaarrfk — APPROVED** [review]

**Pull request merged**